CMakeLists.txt

```diff
@@ -243,6 +239,31 @@ if(CUDA_FOUND)
       message(WARNING "On some Unix systems CUDA 10.0+ requires CMake 3.12.2+; you use CMake ${CMAKE_VERSION}")
   endif()
 
+  # We want to compile as many targets as possible but different CUDA versions support different targets.
+  # Let's instead enable options based on what cuda version we have.
```

Comment:
With the very last changes, does this comment need to be updated?

Reply:
No, because Cuda 9.0 enables `35 50, 60 and 70`, while CUDA10 enables `75` on top of it and CUDA11 enables `80` on top of 10. 
When CUDA12 is released, we'd have to drop `35 and 50`.